clippy_lints/src/methods/mod.rs

```diff
@@ -3364,14 +3394,21 @@ fn lint_suspicious_map(cx: &LateContext<'_>, expr: &hir::Expr<'_>) {
     );
 }
 
+const OPTION_AS_REF_DEREF_MSRV: &str = "1.40.0";
+
 /// lint use of `_.as_ref().map(Deref::deref)` for `Option`s
 fn lint_option_as_ref_deref<'tcx>(
     cx: &LateContext<'tcx>,
     expr: &hir::Expr<'_>,
     as_ref_args: &[hir::Expr<'_>],
     map_args: &[hir::Expr<'_>],
     is_mut: bool,
+    msrv: &Option<String>,
 ) {
+    if msrv.is_some() && Version::parse(&msrv.as_ref().unwrap()) < Version::parse(OPTION_AS_REF_DEREF_MSRV) {
```

Comment:
We should put this in a utils function:

```rust
fn meets_msrv(msrv: Option<&str>, lint_msrv: &str) -> bool { 
    if let Some(msrv) = msrv {
        Version::parse(msrv) >= Version::parse(lint_msrv)
    } else {
        true
    }
}
```

And here: 
```suggestion
    if !meets_msrv(msrv.as_ref(), OPTION_AS_REF_DEREF_MSRV) {
```

Reply:
Sounds good!